examples/eks-hybrid-nodes/ami/amazon-eks-ubuntu.pkr.hcl

```diff
@@ -0,0 +1,319 @@
+      filters = try(security_group_filter.value.filters, null)
+    }
+  }
+
+  session_manager_port    = var.session_manager_port
+  shutdown_behavior       = var.shutdown_behavior
+  skip_profile_validation = var.skip_profile_validation
+  source_ami              = data.amazon-parameterstore.this.value
+
+  dynamic "subnet_filter" {
+    for_each = length(var.subnet_filter) > 0 ? [var.subnet_filter] : []
+
+    content {
+      filters   = try(subnet_filter.value.filters, null)
+      most_free = try(subnet_filter.value.most_free, null)
+      random    = try(subnet_filter.value.random, null)
+    }
+  }
+
+  subnet_id = var.subnet_id
+
+  dynamic "temporary_iam_instance_profile_policy_document" {
+    for_each = length(var.temporary_iam_instance_profile_policy_document) > 0 ? [var.temporary_iam_instance_profile_policy_document] : []
+
+    content {
+      dynamic "Statement" {
+        for_each = temporary_iam_instance_profile_policy_document.value
+
+        content {
+          Action   = try(Statement.value.Action, [])
+          Effect   = try(Statement.value.Effect, "Allow")
+          Resource = try(Statement.value.Resource, ["*"])
+        }
+      }
+      Version = "2012-10-17"
+    }
+  }
+
+  temporary_security_group_source_cidrs     = var.temporary_security_group_source_cidrs
+  temporary_security_group_source_public_ip = var.temporary_security_group_source_public_ip
+  user_data                                 = var.user_data
+  user_data_file                            = var.user_data_file
+
+  dynamic "vpc_filter" {
+    for_each = length(var.vpc_filter) > 0 ? var.vpc_filter : []
+
+    content {
+      filters = try(vpc_filter.value.filters, null)
+    }
+  }
+
+  vpc_id = var.vpc_id
+
+  dynamic "metadata_options" {
+    for_each = length(var.metadata_options) > 0 ? [var.metadata_options] : []
+
+    content {
+      http_endpoint               = try(metadata_options.value.http_endpoint, null)
+      http_put_response_hop_limit = try(metadata_options.value.http_put_response_hop_limit, null)
+      http_tokens                 = try(metadata_options.value.http_tokens, null)
+      instance_metadata_tags      = try(metadata_options.value.instance_metadata_tags, null)
+    }
+  }
+}
+
+################################################################################
+# Build
+################################################################################
+
+build {
+  sources = ["source.amazon-ebs.this"]
+
+  provisioner "shell" {
+    execute_command = "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'"
+
+    env = {
+      DEBIAN_FRONTEND = "noninteractive"
+    }
+
+    expect_disconnect = true
+
+    inline = [
+      "cloud-init status --wait",
+      "apt update",
+      "apt upgrade -y",
+      "apt install iptables conntrack -y",
+      "systemctl reboot",
+    ]
+
+    pause_after = "15s"
+  }
+
+  provisioner "shell" {
+    execute_command = "echo 'packer' | sudo -S sh -c '{{ .Vars }} {{ .Path }}'"
```

Comment:
Should it really begin with - echo packer? Why not start it with sudo and not use pipe?

Reply:
its Packer's way of executing commands as sudo when using a non-root role https://developer.hashicorp.com/packer/docs/provisioners/shell#sudo-example